tests/tel2puml/otel_to_pv/test_sequence_otel_v2.py

```diff
@@ -602,3 +603,155 @@ def mock_fetch_data_holder(config: IngestDataConfig) -> SQLDataHolder:
         assert all(isinstance(event, OTelEvent) for event in events)
         assert all(count == 2 for count in job_id_count.values())
         assert len(valid_event_ids) == 0
+
+    def test_otel_to_pv(
+        self,
+        monkeypatch: MonkeyPatch,
+        mock_yaml_config_dict: dict[str, Any],
+        sql_data_holder_with_otel_jobs: SQLDataHolder,
+        mock_yaml_config_string: str,
+        mock_temp_dir_with_json_files: Path,
+    ) -> None:
+        """Tests for the function otel_to_pv."""
+
+        # Test 1: Default parameters
+        def mock_fetch_data_holder(config: IngestDataConfig) -> SQLDataHolder:
+            return sql_data_holder_with_otel_jobs
+
+        ingest_data_config = IngestDataConfig(
+            data_sources=mock_yaml_config_dict["data_sources"],
+            data_holders=mock_yaml_config_dict["data_holders"],
+            ingest_data=IngestTypes(**mock_yaml_config_dict["ingest_data"]),
+        )
+        monkeypatch.setattr(
+            "tel2puml.otel_to_pv.sequence_otel_v2.fetch_data_holder",
+            mock_fetch_data_holder,
+        )
+
+        result = otel_to_pv(ingest_data_config)
+
+        events = []
+        valid_job_names = {"test_name"}
+        valid_event_ids = [f"{i}_{j}" for i in range(5) for j in range(2)]
+        valid_job_ids = {f"test_id_{i}" for i in range(5)}
+        job_id_count: dict[str, int] = {}
+        for job_name, pv_event_streams in result:
+            assert job_name in valid_job_names
+            for pv_event_gen in pv_event_streams:
+                for pv_event in pv_event_gen:
+                    job_id_count.setdefault(pv_event["jobId"], 0)
+                    job_id_count[pv_event["jobId"]] += 1
+                    events.append(pv_event)
+                    assert pv_event["eventId"] in valid_event_ids
+                    assert pv_event["jobId"] in valid_job_ids
+                    valid_event_ids.remove(pv_event["eventId"])
+
+        assert len(events) == 10
+        assert all(count == 2 for count in job_id_count.values())
+        assert len(valid_event_ids) == 0
+
+        # Test 2: ingest_data = True
+        config = self.get_ingest_config(
+            mock_yaml_config_string, mock_temp_dir_with_json_files
+        )
+        result = otel_to_pv(config, ingest_data=True)
+
+        events = []
+        valid_job_names = {"Backend TestJob", "Frontend TestJob"}
```

Comment:
Again with a set I can't see how many times you are expecting each job name (I assume two but from the tests I can't tell)